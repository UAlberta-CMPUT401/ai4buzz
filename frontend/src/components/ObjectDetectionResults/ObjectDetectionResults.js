import React from 'react';

import styles from './ObjectDetectionResults.module.css';

const ObjectDetectionResults = ({ results }) => {
  return (
    <>
      {Object.keys(results).map((object, idx) => {
        // ignore the bounding box image string
        if (object === 'processes_bounding_boxes_image_as_base64_string')
          return null;
        return (
          <div key={idx} className={styles.objectDetectionResultsData}>
            <div>{object}</div>
            <div>
              {results[object].confidences.map((confidence, idx) => {
                return (
                  <div className="value" key={idx}>
                    {(confidence * 100).toFixed(3)}%
                  </div>
                );
              })}
            </div>
          </div>
        );
      })}
    </>
  );
};

export default ObjectDetectionResults;
