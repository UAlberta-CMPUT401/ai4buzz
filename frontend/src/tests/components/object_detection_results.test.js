import React from 'react';
import ObjectDetectionResults from '../../components/ObjectDetectionResults/ObjectDetectionResults';
import renderer from 'react-test-renderer';

it('Renders ImageClassification component correctly', () => {
  const objectDetectionResults = {
    person: {
      freq: 2,
      confidences: [0.9994099140167236, 0.9985186457633972],
    },
  };
  const tree = renderer
    .create(<ObjectDetectionResults results={objectDetectionResults} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
