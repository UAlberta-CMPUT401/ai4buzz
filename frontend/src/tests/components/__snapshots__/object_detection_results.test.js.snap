// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ImageClassification component correctly 1`] = `
<div
  className="objectDetectionResultsData"
>
  <div>
    person
  </div>
  <div>
    <div
      className="value"
    >
      99.941
      %
    </div>
    <div
      className="value"
    >
      99.852
      %
    </div>
  </div>
</div>
`;
